spec/services/theme_settings_migrations_runner_spec.rb

```diff
@@ -0,0 +1,285 @@
+    end
+
+    it "raises a clear error message when the migration file doesn't export anything" do
+      migration_field.update!(value: <<~JS)
+        function migrate(settings) {
+          return settings;
+        }
+      JS
+
+      expect do described_class.new(theme).run end.to raise_error(
+        Theme::SettingsMigrationError,
+        I18n.t("themes.import_error.migrations.no_exported_function", name: "0001-some-name"),
+      )
+    end
+
+    it "raises a clear error message when the migration file exports the default as something that's not a function" do
+      migration_field.update!(value: <<~JS)
+        export function migrate(settings) {
+          return settings;
+        }
+
+        const AA = 1;
+        export default AA;
+      JS
+
+      expect do described_class.new(theme).run end.to raise_error(
+        Theme::SettingsMigrationError,
+        I18n.t(
+          "themes.import_error.migrations.default_export_not_a_function",
+          name: "0001-some-name",
+        ),
+      )
+    end
+
+    it "raises a clear error message when the migration function doesn't return anything" do
+      migration_field.update!(value: <<~JS)
+        export default function migrate(settings) {}
+      JS
+
+      expect do described_class.new(theme).run end.to raise_error(
+        Theme::SettingsMigrationError,
+        I18n.t("themes.import_error.migrations.no_returned_value", name: "0001-some-name"),
+      )
+    end
+
+    it "raises a clear error message when the migration function doesn't return a Map" do
+      migration_field.update!(value: <<~JS)
+        export default function migrate(settings) {
+          return {};
+        }
+      JS
+
+      expect do described_class.new(theme).run end.to raise_error(
+        Theme::SettingsMigrationError,
+        I18n.t("themes.import_error.migrations.wrong_return_type", name: "0001-some-name"),
+      )
+    end
+
+    it "surfaces runtime errors that occur within the migration" do
+      migration_field.update!(value: <<~JS)
+        export default function migrate(settings) {
+          null.toString();
+          return settings;
+        }
+      JS
+
+      expect do described_class.new(theme).run end.to raise_error(
+        Theme::SettingsMigrationError,
+        I18n.t(
+          "themes.import_error.migrations.runtime_error",
+          name: "0001-some-name",
+          error: "TypeError: Cannot read properties of null (reading 'toString')",
+        ),
+      )
+    end
+
+    it "returns a list of objects that each has data representing the migration and the results" do
+      results = described_class.new(theme).run
+
+      expect(results[0][:version]).to eq(1)
+      expect(results[0][:name]).to eq("some-name")
+      expect(results[0][:original_name]).to eq("0001-some-name")
+      expect(results[0][:theme_field_id]).to eq(migration_field.id)
+      expect(results[0][:settings_before]).to eq({})
+      expect(results[0][:settings_after]).to eq({})
```

Comment:
I think it is important to test this for various settings type as well as multiple settings as well.